Documentation for the Authorization filter (#2992)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/authorization/con-authorization-model.adoc

:_mod-docs-content-type: CONCEPT

[id='con-authorization-model-{context}']
= Authorization Model

In Kafka, clients perform operations on resources in Kafka.

> The tables below lists the resource types and the operations that apply to them.

Reviewer: ```suggestion
The following tables list the resource types and the operations that apply to them.
```

Reviewer: A note that the filter currently supports only topics and the plan is to support more resources would be useful. Or similar, depending on the plan

Author: yes, the is for other resource types to be covered too.

> == Resource type and operations

Reviewer: ```suggestion
== Resource types and operations
```

This table lists the resources types and operations enforced by the authorization filter.

[id='authorization-resource-types-and-operations-{context}']
.Table of authorization resource types and operation
|===
|Resource type|Operations|Typical use-case

.8+|Topic
|READ
|Required for a consuming client to fetch record.

|WRITE
|Required for a producing client to produce records.
|CREATE
.3+|Required for an admin client to create, delete or alter topics.

|DELETE

|ALTER

|DESCRIBE
|Required for an admin client to perform the describe operations that refer to topic resources.

|DESCRIBE_CONFIGS
|Required for an admin client to perform describe config operations that refer to topic configuration.

|ALTER_CONFIGS
|Required for an admin client to perform alter config operations that relate to topic configuration.

// |Cluster
// |CREATE, ALTER, DESCRIBE, CLUSTER_ACTION, DESCRIBE_CONFIGS, ALTER_CONFIGS

|===

== Implied Operations

In the authorization model, having permissions to some, _higher_, operations implies permissions on other, _lower_, operations.

[id='authorization-implied-operations-{context}']

> .Table of implied operations

Reviewer: ```suggestion

```

:_mod-docs-content-type: CONCEPT

[id='con-authorization-primitives-{context}']
= Authorization Primitives

Kafka clients send Kafka messages to perform operations on resources in Kafka.
It is required to know the operations and resources to effectively configure authorization checks.
This section lists these operations and resources, and then list the mapping between the Kafka messages and the operations and resources.

== Resource type and operations

This table lists the resources types and operations enforced by the authorization filter.

[id='authorization-resource-types-and-operations-{context}']
.Table of authorization primitives
|===
|Resource type|Operations

|Topic
|READ, WRITE, CREATE, DELETE ALTER, DESCRIBE, DESCRIBE_CONFIGS, ALTER_CONFIGS

|Cluster
|CREATE, ALTER, DESCRIBE, CLUSTER_ACTION, DESCRIBE_CONFIGS, ALTER_CONFIGS

|===

== Message, operation, and resource mapping

The table below lists the kafka messages for which the authorization filter will make authorization enforcements.
For each message, it shows the operation and resource type that will be tested.

[id='authorization-message-mapping-{context}']

> .Table mapping the message to operation and resource

Author: Agreed, I was having second thoughts myself.  I've removed the table.
